libraries/abstractions/common_io/include/iot_sdio.h

```diff
@@ -0,0 +1,709 @@
+                                       /*!< 1U is reserved per SDIO specification*/
+#define IOT_SDIO_BUS_4BIT   ( 2 )      /*!< 4 bit bus mode */
+#define IOT_SDIO_BUS_8BIT   ( 3 )      /*!< 8 bit bus mode */
+
+/**
+ * @brief sdio io read/write direction
+ */
+typedef enum IotSdioDirection
+{
+    eIORead = 0U,  /*!< io read */
+    eIOWrite = 1U, /*!< io write */
+} IotSdioDirection_t;
+
+/**
+ * @brief SDIO I/O function number type
+ *        Each SDIO card can support up to 7 I/O functions.
+ *        Each I/O function can implement one of the following:
+ *          - SDIO Standard UART
+ *          - SDIO Bluetooth Type-B standard interface
+ *          - SDIO GPS standard interface
+ *          - SDIO Camera standard interface
+ *          - SDIO PHS standard interface
+ *          - SDIO WLAN interface
+ *          - Embedded SDIO-ATA standard interface
+ *          - SDIO Bluetooth Type-A AMP standard interface
+ *          - Other SDIO standard inferface type identified by
+ *            Extended SDIO Standard Function interface code
+ */
+typedef enum IotSdioFunction
+{
+    eSdioFunction0,       /*!< refer to common register area present on all I/O cards*/
+    eSdioFunction1,       /*!< refer to register area of 1st I/O function if card supports it*/
+    eSdioFunction2,       /*!< refer to register area of 2nd I/O function if card supports it*/
+    eSdioFunction3,       /*!< refer to register area of 3rd I/O function if card supports it*/
+    eSdioFunction4,       /*!< refer to register area of 4th I/O function if card supports it*/
+    eSdioFunction5,       /*!< refer to register area of 5th I/O function if card supports it*/
+    eSdioFunction6,       /*!< refer to register area of 6th I/O function if card supports it*/
+    eSdioFunction7,       /*!< refer to register area of 7th I/O function if card supports it*/
+    eSdioFunctionMemory,  /*!< refer to register area of memory function of a combo card*/
+} IotSdioFunction_t;
+
+/**
+ * @brief card detect type
+ */
+typedef enum IotSdioCardDetectType
+{
+    eDetectCardByGpioCD,    /*!< sdio card detect by CD pin through GPIO */
+    eDetectCardByHostCD,    /*!< sdio card detect by CD pin through host */
+    eDetectCardByHostDATA3, /*!< sdio card detect by DAT3 pin through host */
+    eDetectCardNonStandard, /*!< sdio card detect by non standard method
+                                 certain sdio host may support */
+} IotSdioCardDetectType_t;
+
+/**
+ * @brief common notify event types in sdio callback
+ */
+typedef enum IotSdioEventType
+{
+    eSdioCardInterruptEvent,    /*!< Interrupt received from sdio card */
+    eSdioCardInsertedEvent,     /*!< sdio card insertion is detected */
+    eSdioCardRemovedEvent,      /*!< sdio card removal is detected */
+    eSdioCardPowerOnEvent,      /*!< Powering on sdio card */
+    eSdioCardPowerOffEvent,     /*!< powering off sdio card */
+} IotSdioEventType_t;
+
+/**
+ * @brief Ioctl request types.
```

Comment:
Inconsistent capitalization of IOCTL and/or the @brief comment.

Reply:
Capitalized IOCTL